wandb/sdk_py27/internal/internal_util.py

```diff
@@ -85,9 +88,15 @@ def _process(self, record):
     def _finish(self):
         raise NotImplementedError
 
+    def _debounce(self):
+        raise NotImplementedError
+
     def _run(self):
         self._setup()
+        start = time.time()
         while not self._stopped.is_set():
+            if time.time() - start >= self._debounce_interval_ms / 1000.0:
+                self._debounce()
```

Comment:
Are we properly handling the final sync?  I.E. `self._stopped` is set, we should bypass the debounce.

Reply:
no need, no time has passed since the last check.
stop is a signal that this should stop, the final sync happens in the defer statemachine